_minutes/2024-09-24-wecg-tpac.md

```diff
@@ -0,0 +1,466 @@
+ * [patrick] Maybe names like chrome.thisWillGoAwayBySomeDate
+ * [devlin] Could incorporate the second bucket into the API namespace.
+ * [jordan] How about “unstable”?
+ * [devlin] Good candidate. Exact name does not matter. Can bikeshed async.
+ * [devlin] Do we want to use the separate bucket idea for this?
+ * [simeon] And this is independent of yesterday's discussion on origin trials, right?
+ * [devlin] Yes, yesterday is about APIs that we do not have the intention of breaking, this is about APIs that we do not want to ship to stable.
+ * [rob] To clarify, this is for features that we don't want to ship to stable?
+ * [devlin] Not mainline, may still reach stable release channel.
+ * [rob] What would be the extension store policy for accepting these?
+ * [devlin] Store decision per API. There may be some that we want to expose publicly. That's part of why we can't use experimental – the experimental permission is auto-rejected.
+ * [rob] When I asked my team about names, “trials” was one of the suggestions for the namespace. It is not verbose though.
+ * [devlin] And may be confused with origin trials. I prefer alternatives such as “unstable”, “exploratory”.
+ * [devlin] Tentative decision: use a namespace (name to be selected) for experimental APIs. I'll take the action item to write this up.
+ * [devlin] Would other browsers expect adoption?
+ * [timothy] Yes, would be useful.
+ * [rob] Yes.
+ * [devlin] To what extent do we want to raise these in the WECG? On Chrome's side it may be a FYI.
+ * [timothy] Heads-up would be good. E.g. “Do you have plans for names in this road”, to avoid sidebar/sidepanel issue.
+ * [david] Since we are thinking of this as likely not conflicting, do we want to think this as reserving a namespace where each browser can do something on their own?
+ * [devlin] That's what it will likely end up being. Hopeful we won't collide, but should communicate to devs that if it happens it doesn't necessarily mean anything.
+ * [simeon] A challenge there is that there is no reliable way to distinguish browsers.
+ * [devlin] Almost no developer should be using this API unless they really want to use this API. Separately, I appreciate the challenge of identifying the browser, we should discuss how to address it later.
+ * [carlos] Reliable way is to check the protocol of the origin (chrome-extension, moz-extension, etc.)
+ * [oliver] Even more so, between browser engines we should give heads-up. E.g. no two Chromium forks with the same name.
+ * [devlin] My stance is that if extension developers are using these APIs, they should exactly know what they are doing. It should target specific browsers.
+ * [patrick] The idea is that these APIs would not be added to main extensions.
+ * [rob] As an extension developer, I often look for a way to accomplish my goal and will use whatever API is available to accomplish it.
+ * [patrick] You can shave your face with a chainsaw, but that doesn't make it a good idea.
```

Comment:
I wonder if we should drop this line. I'm not sure if it needs to be in the notes 😅